### PR TITLE
clean up code, added comments, fixed other things dealing with coding…

### PR DESCRIPTION
**Pull request by Joost001**
… standards

### CODE REVIEW

**Comment by jovasque156** [comment]
There are many changes in code I wrote. I suggest to add me as reviewers.

File: src/encoder_decoder.py [hunk 1]
```diff
@@ -1,138 +1,139 @@
...
+        :param queries:
+        :param keys:
+        :param values:
+        :param mask:
+        :return:
+        """
+        # Get number of training examples
+        n_batch = queries.size()[0]
+
         queries, keys, values = \
```
Comment (jovasque156) [inline src/encoder_decoder.py:73]: Take care with this line. It doesn't work when model and data are transferred to cuda. I guess the problem is in line 22. I suggest to create a layer for each input (queries, keys, values) separately. Like this: 

self.linear_queries = nn.Linear(embed_size, embed_size)
self.linear_keys = nn.Linear(embed_size, embed_size)
self.linear_values = nn.Linear(embed_size, embed_size)

Comment (jonathankabala) [reply src/encoder_decoder.py:73]: yes

**Comment by jovasque156** [comment]
I'm fine the modifications on the code I wrote. Thank you for clean and organize better the code.

File: src/encoder_decoder.py (hunk 1 above)
Comment (Joost001) [reply src/encoder_decoder.py:73]: Line 22 or 71?

**Comment by jonathankabala** [comment]
I think what Jonathan suggested is fine. The rest is fine too. If we can modify that, then we can merge

File: src/encoder_decoder.py (hunk 1 above)
Comment (jonathankabala) [reply src/encoder_decoder.py:73]: I think it's replacing line 32 with what Jonathan V suggests. Which will make you change lines 68-70

Comment (jovasque156) [reply src/encoder_decoder.py:73]: What Jonathan B says. You will need to change line 32 (22) and then lines 68-70 (73-75).

**Pull request merged**